File: app/src/main/java/ca/warp7/android/scouting/resources/ActionVibrator.java
package ca.warp7.android.scouting.resources;

import android.content.Context;
import android.os.Vibrator;

import ca.warp7.android.scouting.abstraction.AbstractActionVibrator;

/**
 * @since v0.4.1
 */

public class ActionVibrator implements AbstractActionVibrator {
    private Vibrator mActualVibrator;
    private boolean mVibrationOn;

    ActionVibrator(Context context, boolean vibrationOn) {
        mActualVibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
        mVibrationOn = vibrationOn;
    }

    @Override
    public void vibrateStart() {
        if (mVibrationOn) {
            mActualVibrator.vibrate(kStartVibration, -1);
        }
    }

    @Override
    public void vibrateAction() {
        if (mVibrationOn) {
            mActualVibrator.vibrate(kActionEffectVibration);
        }
    }

    private static final long[] kStartVibration = new long[]{0, 20, 30, 20};
    private static final int kActionEffectVibration = 30;
}

File: app/src/main/java/ca/warp7/android/scouting/res/ActionVibrator.kt
package ca.warp7.android.scouting.res

import android.content.Context
import android.os.Vibrator
import ca.warp7.android.scouting.AbstractActionVibrator

/**
 * @since v0.4.1
 */

class ActionVibrator(context: Context,
                     private val mVibrationOn: Boolean) : AbstractActionVibrator {

    private val actual = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator

    override fun vibrateStart() {
        if (mVibrationOn) {
            actual.vibrate(kStartVibration, -1)
        }
    }

    override fun vibrateAction() {
        if (mVibrationOn) {
            actual.vibrate(kActionEffectVibration.toLong())
        }
    }

    companion object {
        private val kStartVibration = longArrayOf(0, 20, 30, 20)
        private const val kActionEffectVibration = 30
    }
}
